R/areaplot.R

```diff
@@ -141,9 +142,17 @@ AreaPlot <- function(
         datas <- split(data, data[[split_by]])
         # keep the order of levels
         datas <- datas[levels(data[[split_by]])]
+        palette <- check_palette(palette, names(datas))
+        palcolor <- check_palette(palcolor, names(datas))
     } else {
         datas <- list(data)
+        palette <- list(palette)
         names(datas) <- "..."
+        names(palette) <- "..."
+        if (!is.null(palcolor)) {
+	        palcolor <- list(palcolor)
+            names(palcolor) <- "..."
+        }
```

Comment:
Maybe also check them here to ensure there is only a single element? Maybe reuse the `check_palette` function (`check_palette(palette, palcolor, "...")`)?